api/src/deployment.rs

```diff
@@ -344,11 +363,29 @@ pub(crate) struct Context {
 impl DeploymentSystem {
     pub(crate) async fn new(build_system: Box<dyn BuildSystem>, fqdn: String) -> Self {
         let router: Arc<Router> = Default::default();
+        let (tx, rx) = std::sync::mpsc::sync_channel::<Log>(64);
 
         let deployments = Arc::new(RwLock::new(
             Self::initialise_from_fs(&build_system.fs_root(), &fqdn).await,
         ));
 
+        let deployments_log = deployments.clone();
+
+        tokio::spawn(async move {
+            loop {
```

Comment:
It should, although the rust compiler / lifetime checker is not getting the correct lifetime of the the `rx` when using `while let`. Since the `while let` will have an internal `await` and the `rx` is no `Send`, the compiler complains the `rx` cannot be send at the `await`